common/app/model/Cached.scala

```diff
@@ -29,41 +31,55 @@ object Cached extends implicits.Dates {
 
   private val tenDaysInSeconds = 864000
 
-  def apply(seconds: Int)(result: Result): Result = {
-    if (cacheableStatusCodes.contains(result.header.status)) cacheHeaders(seconds, result) else result
-  }
-
-  def apply(duration: Duration)(result: Result): Result = {
-    apply(duration.toSeconds.toInt)(result)
-  }
+  case class Hash(string: String)
 
-  def apply(page: Page)(result: Result): Result = {
-    val cacheSeconds = page.metadata.cacheTime.cacheSeconds
-    if (cacheableStatusCodes.contains(result.header.status)) cacheHeaders(cacheSeconds, result) else result
-  }
+  sealed trait CacheableResult
+  case class RevalidatableResult(result: Result, hash: Hash) extends CacheableResult
+  case class WithoutRevalidationResult(result: Result) extends CacheableResult
 
-  case class Hash(string: String)
-  case class RevalidatableResult(result: Result, hash: Hash)
   object RevalidatableResult {
-    def apply(result: Result, body: String) = {
+    def apply[C](result: Result, content: C)(implicit writeable: Writeable[C]) = {
       // hashing function from Arrays.java
-      val hashLong: Long = body.getBytes("UTF-8").foldLeft(z = 1L){
+      val hashLong: Long = writeable.transform(content).foldLeft(z = 1L){
         case (accu, nextByte) => 31 * accu + nextByte
       }
       new RevalidatableResult(result, Hash(hashLong.toString))
     }
+
+    def Ok[C](content: C)(implicit writeable: Writeable[C]) = {
+      apply(Results.Ok(content), content)
+    }
+  }
```

Comment:
Do we need to support more result status code? I mean not only 200 OK but also other code from the 2XX family